src/icon.c

```diff
@@ -163,11 +163,11 @@ cairo_surface_t *icon_get_for_notification(const notification *n)
                 if (w >= h) {
                         scaled = gdk_pixbuf_scale_simple(pixbuf,
                                         settings.max_icon_size,
-                                        (int) ((double) settings.max_icon_size / w * h),
```

Comment:
You're changing logic here and not just copying the code from file a to b. IMHO I'd still separate it into a single commit.

Reply:
You're right, I changed it to add it as a fixup to the the commit that removed some of the previous floating-point hackery that existed in `draw.c`.